### PR TITLE
Convert var to let & Convert Non-strict to strict equality checking

### PR DESCRIPTION
**Pull request by kale-io**

### CODE REVIEW

File: packages/gatsby-react-router-scroll/src/StateStorage.js [hunk 1]
```diff
@@ -48,6 +48,6 @@ export default class SessionStorage {
   getStateKey(location, key) {
     const locationKey = location.key
     const stateKeyBase = `${STATE_KEY_PREFIX}${locationKey}`
-    return key == null ? stateKeyBase : `${stateKeyBase}|${key}`
+    return key === null ? stateKeyBase : `${stateKeyBase}|${key}`
```
Comment (KyleAMathews) [inline packages/gatsby-react-router-scroll/src/StateStorage.js]: This changes the logic as it's not now checking for `undefined` https://stackoverflow.com/a/5515385/182702 If you want to use the strict equality check, could you also add a check for undefined?

Comment (andreipfeiffer) [reply packages/gatsby-react-router-scroll/src/StateStorage.js]: Sure, will do, thanks @KyleAMathews

**Comment by KyleAMathews** [comment]
Thanks!

**Comment by kale-io** [comment]
No problem, glad to help.

**Pull request merged**